### DISCUSSION TITLE
Connect-PNPOnline not working on azure web jobs with client id and client secret

### DISCUSSION THREAD

**Discussion by kaushalbcactus** [pnp/powershell#1738] [General]
Hi Connect-PNPOnline displays the below error on azure webjobs with clientid clientsecret command.
 
![image](https://user-images.githubusercontent.com/49706885/155207396-6071a5c9-0876-49b3-8223-faa90265a17a.png)

below is code
`$path = "C:\home\site\wwwroot"
$env:PSModulePath += ";$path\Modules\PnP.PowerShell"
Connect-PnPOnline -Url $SiteURL -ClientId "xxx" -ClientSecret "xxx" 
Get-PnPContext
`
Also same issue with 
`New-AzStorageContext`

1. When I replace clientid and clientsecret with emailid of user and encrypted password then it works perfectly in azure webjobs.
2. Above code works in Azure functions and local powershell but fails in azure web jobs

Requesting your suggestion to resolve the issue.
Thanks!

**Comment by gautamdsheth** [comment]
Hi am converting this to a discussion , not exactly an issue with PnP PowerShell but most likely some configuration issue :(

**Comment by deadgeek** [comment]
I'm having a similar issue. Though for me, when using ClientID and ClientSecret, I can connect. Subsequent PnP commands however throw the following:
Get-PnPUser: Exception has been thrown by the target of an invocation.
